plugins/janus_videoroom.c

```diff
@@ -2024,16 +2059,21 @@ void janus_videoroom_incoming_data(janus_plugin_session *handle, char *buf, int
 	janus_videoroom_session *session = (janus_videoroom_session *)handle->plugin_handle;
 	if(!session || g_atomic_int_get(&session->destroyed) || session->participant_type != janus_videoroom_p_type_publisher)
 		return;
-	janus_videoroom_publisher *participant = (janus_videoroom_publisher *)session->participant;
-	if(participant == NULL || g_atomic_int_get(&participant->destroyed))
+	janus_videoroom_publisher *participant = janus_videoroom_session_get_publisher(session);
+	if(participant == NULL || g_atomic_int_get(&participant->destroyed)) {
+		g_clear_pointer(&participant, janus_videoroom_publisher_dereference);
```

Comment:
`&participant` is a pointer to the local variable that may contain the value `NULL`.  `participant == NULL` tests whether the local variable contains the value `NULL`.